docs/_docs/reference/contextual/derivation-macro.md

```diff
@@ -21,185 +17,175 @@ trait Eq[T]:
-```
+Shown below is the implementation of `summonInstances` as a macro, which for each type `elem` in the tuple type, calls
+`deriveOrSummon[T, elem]`.
 
-One additional difference with the body of `derived` here as opposed to the one
-with `inline` is that with macros we need to synthesize the body of the code during the
-macro-expansion time. That is the rationale behind the `eqProductBody` function.
-Assuming that we calculate the equality of two `Person`s defined with a case
-class that holds a name of type [`String`](https://scala-lang.org/api/3.x/scala/Predef$.html#String-0)
-and an age of type `Int`, the equality check we want to generate is the following:
+To understand `deriveOrSummon`, consider that if `elem` derives from the parent `T` type, then it is a recursive derivation.
+Recursive derivation usually happens for types such as `scala.collection.immutable.::`. If `elem` does not derive from `T`, then there must exist a contextual `Eq[elem]` instance.
 
 ```scala
-   true
-   && Eq[String].eqv(x.productElement(0),y.productElement(0))
-   && Eq[Int].eqv(x.productElement(1), y.productElement(1))
-```
-
-## Calling the derived method inside the macro
+def summonInstances[T: Type, Elems: Type](using Quotes): List[Expr[Eq[_]]] =
+  Type.of[Elems] match
+    case '[elem *: elems] => deriveOrSummon[T, elem] :: summonInstances[T, elems]
+    case '[EmptyTuple]    => Nil
 
-Following the rules in [Macros](../metaprogramming/metaprogramming.md) we create two methods.
-One that hosts the top-level splice `eqv` and one that is the implementation.
-Alternatively and what is shown below is that we can call the `eqv` method
-directly. The `eqGen` can trigger the derivation.
+def deriveOrSummon[T: Type, Elem: Type](using Quotes): Expr[Eq[Elem]] =
+  Type.of[Elem] match
+    case '[T] => deriveRec[T, Elem]
+    case _    => '{ summonInline[Eq[Elem]] }
 
-```scala
-extension [T](inline x: T)
-   inline def === (inline y: T)(using eq: Eq[T]): Boolean = eq.eqv(x, y)
-
-inline given eqGen[T]: Eq[T] = ${ Eq.derived[T] }
+def deriveRec[T: Type, Elem: Type](using Quotes): Expr[Eq[Elem]] =
+  import quotes.reflect.*
+  Type.of[T] match
+    case '[Elem] => report.errorAndAbort("infinite recursive derivation")
+    case _       => derivedMacro[Elem] // recursive derivation
 ```
 
-Note, that we use inline method syntax and we can compare instance such as
-`Sm(Person("Test", 23)) === Sm(Person("Test", 24))` for e.g., the following two
-types:
+One additional difference with the body of `derivedMacro` here as opposed to the one with `inline` is that with macros we need to synthesize the body of the code during the macro-expansion time.
+That is the rationale behind the `eqProductBody` function.
+Assuming that we calculate the equality of two `Person`s defined with a case class that holds a name of type [`String`](https://scala-lang.org/api/3.x/scala/Predef$.html#String-0) and an age of type `Int`, the equality check we want to generate is the following:
```

Comment:
This sentence and the code snippet below should stay higher up in the explanation. It should be *before* the macro code that generates it.